### PR TITLE
Merge feature/button to master

### PR DESCRIPTION
**Pull request by mmroz**
This PR merges the feature/button branch in to master.

I updated the demo app to showcase V1, V2 and button more clearly. There is now an edit button to select the preferred checkout method.

https://github.com/afterpay/sdk-ios/assets/8743061/e6d9493a-ea21-400e-ae86-80cc2718248a

### CODE REVIEW

File: Example/Example/Purchase/CartViewController.swift [hunk 1]
```diff
@@ -93,24 +114,62 @@ final class CartViewController: UIViewController, UITableViewDataSource {
...
+    }
   }
 
   // MARK: Actions
 
   @objc private func didTapPay() {
-    eventHandler(.didTapPay)
+    switch checkoutOption {
+    case .v1Checkout:
+      print("TODO - FIXME")
```
Comment (mmroz) [inline Example/Example/Purchase/CartViewController.swift:155]: @ScottAntonacAP how can I present V1 checkout?

File: Example/Example/Purchase/CartViewController.swift [hunk 2]
```diff
@@ -20,13 +19,29 @@ final class CartViewController: UIViewController, UITableViewDataSource {
   private let titleSubtitleCellIdentifier = String(describing: TitleSubtitleCell.self)
   private let eventHandler: (Event) -> Void
 
+  private var checkoutOption: CheckoutPickerOption = .v1Checkout {
+    didSet {
+      updateViewState()
+    }
+  }
+
   private lazy var cashButton = CashAppPayButton(size: .large) { [weak self] in
```
Comment (mmroz) [inline Example/Example/Purchase/CartViewController.swift:28]: It's not clear to me how to enable this button? @ScottAntonacAP

File: Example/Example/Purchase/CartViewController.swift (hunk 1 above)
Comment (ScottAntonacAP) [reply Example/Example/Purchase/CartViewController.swift:155]: Currently on `main` branch, the express checkout switch in the example app is a actually a V1 / V2 switch. What it should be doing is setting the checkout mode.

That switch (just the express switch) could potentially be removed for now and the functionality moved to the new buttons created in this PR.
 
It _currently_ works as follows:

1. Executing `didTapPay` in `PurchaseFlowController` calls `logicController.payWithAfterpay()`
2. `payWithAfterpay()` checks if express is enabled. if not, run the `showAfterpayCheckoutV1` command with the retrieved checkout url
3. The command from step 2 runs `Afterpay.presentCheckoutModally()` which is the SDK method

In the future functionality could be added to V2 to use Express OR Standard checkout.

File: Example/Example/Purchase/CartViewController.swift (hunk 2 above)
Comment (ScottAntonacAP) [reply Example/Example/Purchase/CartViewController.swift:28]: There is room for improvement in the readability here.

Currently when the CartView loads, this line executes: `eventHandler(.cartDidLoad(self.cashButton))`. Follow this to the `PurchaseFlowController` which then runs `logicController.retrieveCashAppToken(cashButton: button)`. This runs `setCashButtonEnabled(false)`.

When `CashAppPayState` is in a ready to auth state, it then runs `setCashButtonEnabled(true)`

**Review by ScottAntonacAP — APPROVED** [review]
This looks great. 👏 
Also tested locally and all checkout methods work

✅ V1
✅ V2
✅ V2 with CAP
✅ V3
✅ V3 with CAP

**Pull request merged**